src/WPF/WPF.Viewer/Samples/Layers/AddVectorTiledLayerFromCustomStyle/AddVectorTiledLayerFromCustomStyle.xaml.cs

```diff
@@ -0,0 +1,134 @@
+        instructions: "Pan and zoom to explore the vector tile basemap.",
+        tags: new[] { "tiles", "vector", "vector basemap", "vector tiled layer", "vector tiles" })]
+    [ArcGIS.Samples.Shared.Attributes.OfflineData()]
+    public partial class AddVectorTiledLayerFromCustomStyle
+    {
+        // List all portal item IDs of vector tiled layers.
+        private List<string> _itemIDs;
+
+        // These layers can be selected from the ComboBox.
+        private List<ArcGISVectorTiledLayer> _vectorTiledLayers;
+
+        private Viewpoint _defaultViewpoint = new Viewpoint(10, 5.5, 1e8);
+        private Viewpoint _dodgeCityViewpoint = new Viewpoint(37.76528, -100.01766, 4e4);
+
+        public AddVectorTiledLayerFromCustomStyle()
+        {
+            InitializeComponent();
+            _ = Initialize();
+        }
+
+        private async Task Initialize()
+        {
+            _itemIDs = new List<string>
+            {
+                // Vector tiled layers from ArcGISOnline.
+                "1349bfa0ed08485d8a92c442a3850b06",
+                "bd8ac41667014d98b933e97713ba8377",
+                "02f85ec376084c508b9c8e5a311724fa",
+                "1bf0cc4a4380468fbbff107e100f65a5",
+
+                // A vector tiled layer created by the local VTPK and light custom style.
+                "e01262ef2a4f4d91897d9bbd3a9b1075",
+
+                // A vector tiled layer created by the local VTPK and dark custom style.
+                "ce8a34e5d4ca4fa193a097511daa8855"
+            };
+
+            // Load the default portal.
+            ArcGISPortal portal = await ArcGISPortal.CreateAsync();
+
+            // Create vector tiled layers for each portal item.
+            _vectorTiledLayers = new List<ArcGISVectorTiledLayer>();
+
+            // Store a list of tiled layers in the different styles.
+            foreach (string item in _itemIDs)
+            {
+                PortalItem websceneItem = await PortalItem.CreateAsync(portal, item);
+                ArcGISVectorTiledLayer vectorTiledLayer = new ArcGISVectorTiledLayer(websceneItem);
+                _vectorTiledLayers.Add(vectorTiledLayer);
+            }
+
+            // Create a map with the default style.
+            MyMapView.Map = new Map(SpatialReferences.WebMercator);
+            MyMapView.Map.Basemap.BaseLayers.Add(_vectorTiledLayers[0]);
+
+            await MyMapView.SetViewpointAsync(_defaultViewpoint);
+        }
+
+        private async void StyleChooserSelectionChanged(object sender, System.Windows.Controls.SelectionChangedEventArgs e)
+        {
+            // Determine the style from the combobox.
+            ComboBox styleChooser = sender as ComboBox;
+            ArcGISVectorTiledLayer vectorTiledLayer = _vectorTiledLayers[styleChooser.SelectedIndex];
+
+            // Clear the map of the currently loaded layer.
+            MyMapView.Map.Basemap.BaseLayers.RemoveAt(0);
+
+            if (styleChooser.SelectedIndex <= 3)
```

Comment:
https://github.com/Esri/arcgis-runtime-samples-ios/pull/981/files#diff-7ca078c5888a4ef221318f40f7a155dbb158e121c061f73d6d2219a84743a172R114
Here is how iOS does this with two lists.